### ISSUE TITLE
Can't drag n drop files with Thunar

### ISSUE THREAD

**Issue by pdsouza** [maruos/maruos#25]
Steps to repro:

1. In home dir:
`$ echo "yoruichi rocks!" > test.txt`
2. Using Thunar, drag and drop to Desktop => OK
3. Using Thunar, drag and drop back to $HOME => "Error while copying to ... Destination is read-only"
4. Notice that you can do this just fine in a terminal though.

Specifically affects try to copy a camera photo anywhere else. You can use a normal cp from a terminal perfectly though:

```
maru@jessie:~$ ls -al /mnt/Phone/DCIM/Camera/
total 16688
drwxrwx---. 2 root sdcard_r    4096 Oct 28  1970 .
drwxrwx---. 4 root sdcard_r    4096 Mar 11 05:53 ..
-rw-rw----. 1 root sdcard_r 1784432 Oct 28  1970 IMG_19701028_062726.jpg
-rw-rw----. 1 root sdcard_r  390853 Oct 28  1970 IMG_19701028_062734.jpg
-rw-rw----. 1 root sdcard_r  365744 Oct 28  1970 IMG_19701028_062736.jpg
-rw-rw----. 1 root sdcard_r  962253 Mar 11 05:54 IMG_20160310_215401.jpg
-rw-rw----. 1 root sdcard_r 7192218 Oct 28  1970 VID_19701028_062732.mp4
-rw-rw----. 1 root sdcard_r 7169528 Mar 11 05:54 VID_20160310_215407.mp4
maru@jessie:~$ cp /mnt/Phone/DCIM/Camera/IMG_19701028_062726.jpg ~/Desktop
maru@jessie:~$ ls ~/Desktop/
IMG_19701028_062726.jpg  Phone.desktop  test.txt
```

Looks like this could be a bug in Thunar: https://bugs.launchpad.net/ubuntu/+source/thunar/+bug/1332623.

Seems identical because I can't drag n drop with Thunar...but I can delete it with Thunar!

Jessie runs 1.6.3 which is still affected.

**WORKAROUND**: Use `cp` from terminal

**Comment by dianaxxyyzz** [comment]
I've also tested latest thunar and that have also problems , I have to recompile the debian stable version .
They say "Using Thunar 1.6.10 resolve the problem" but in our case it do not resolve it .

**Comment by dianaxxyyzz** [comment]
Ok ,now is worcking .
 like a quick fix , I edited the /usr/sbin/thunar binary and at this 2 addresses:

000517f0         cmp        r0, #0x0                   //make nop this
000517f2         bne.w      0x51928                    //make nop this

All looks good till now , I tried to copy files and works .You can use the attached binary of thunar, if you do not want to dissansamble yourself the binary .I'm attaching it , just put it over your existing /usr/bin/thunar .I had to kill the previous running thunar before replacing .
[thunar.zip](https://github.com/maruos/maruos/files/586693/thunar.zip)

**Comment by pdsouza** [comment]
Thanks for debugging this @dianaxxyyzz! Impressive debugging work!

I wonder if there's an issue with the way the root filesystem is mounted in the container that makes Thunar think that its read-only?

**Comment by dianaxxyyzz** [comment]
Hi again ,

Yes ,it may be .I did not had time to dig more , I just wanted a quick fix ,to save time.
It may be also a problem somewhere deep in thunar .
I saw a lot of complains from users who had this problems on usb flash drives too , they could copy with cp but not with thunar , so is hard to tell from where problem really comes.
I just wanted to have a working thunar ,I did nto had time to dig more in thuanr code ,but I hope the thunar devs will fix it in future.
Have a great day !
Dionis

**Comment by dianaxxyyzz** [comment]
I have started to port Maru OS to LG v10 , can you please tell me if you did any modifications to the android source ,so it to display the video over HDMI?
I see you add thed the 2 services(programs ) that make a bridge for video over unix sockets ,but I not know if you also modified something in android source ,to make this to work,
Thank you and have a great day!

**Comment by pdsouza** [comment]
Yeah, there are changes to many parts of Android to make everything work. Basically, if you see an "`android_*` repository listed for this organization, then Maru OS has changes there for Android. As long as you use the Maru OS manifest when building your port you should be fine since you will automatically pull those changes, and they are device-independent, except for a few changes in the device repositories themselves. You can look at the device repo for hammerhead as an example of the device-specific changes needed.

**Comment by dianaxxyyzz** [comment]
Hello, 
Thank you!
Can you please let me know if there are any modifications to this ?
android_vendor*
android_hardware*
android_frameworks*
android_external*
I'm inspecting myself maru os source but I just want to be sure .
Thank you and have a great day!

**Comment by pdsouza** [comment]
Hi Dionis, let's move this converation to your LG thread so it is easier to follow: https://groups.google.com/forum/#!topic/maru-os-dev/-K_lysxngD4
